addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/UrlUtils.java

```diff
@@ -0,0 +1,162 @@
+            new HttpSender(
+                    Model.getSingleton().getOptionsParam().getConnectionParam(),
+                    true,
+                    HttpSender.MANUAL_REQUEST_INITIATOR);
+    // private static HttpRequestConfig config =
+    // HttpRequestConfig.builder().setSoTimeout(10).build();
+
+    /**
+     * Returns a List of all parameters to be used from a given wordlist file.
+     *
+     * @param path the path to the wordlist.
+     * @return a List<String> of all parameters to be used from a given wordlist file.
+     */
+    public static List<String> reader(String path) {
+        logger.info("Reading keys from file: " + path);
+        List<String> params = new ArrayList<>();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(path));
+            String line;
+            while ((line = br.readLine()) != null) {
+                params.add(StringUtils.strip(line));
+            }
+            br.close();
+            logger.info("Succesfully parsed keys!");
+        } catch (Exception e) {
+            logger.error(e);
+        }
+        return params;
+    }
+
+    /**
+     * Returns "crashproof" HTTP Responses from a given URL by using GET method.
+     *
+     * @param url the url to be tested.
+     * @param headers the associated headers.
+     * @return a HttpMessage of the HTTP Response from the given URL.
+     */
+    public static HttpMessage stableRequest(String url, HttpRequestHeader headers) {
+        try {
+            URI parsed = new URI(url, true);
+            String scheme = parsed.getScheme();
+            String host = parsed.getHost();
+            String path = parsed.getPath();
+            String port = parsed.getPort() == -1 ? "" : Integer.toString(parsed.getPort());
+            String[] schemes =
+                    (scheme.equalsIgnoreCase("https"))
+                            ? new String[] {"https", "http"}
+                            : new String[] {"http", "https"};
+            int errorCount = 0;
+            for (String s : schemes) {
+                try {
+                    HttpMessage msg = new HttpMessage();
+                    headers.setURI(new URI(s + "://" + host + ":" + port + path, true));
+                    headers.setMethod("GET");
+                    msg.setRequestHeader(headers);
+                    sender.sendAndReceive(msg);
+                    return msg;
+                } catch (Exception e) {
+                    // Check if it is a connection error
+                    if (e.getMessage().contains("ConnectionError")
+                            || e.getMessage().contains("IOException")) {
+                        continue;
+                    }
+                    logger.info(e);
+                    errorCount = errorCount + 1;
+                }
+                logger.info("Total number of Errors [stableRequest()]: " + errorCount);
+            }
+        } catch (URIException e) {
+            logger.error(e);
```

Comment:
This will log just the exception message, was that on purpose?

Reply:
Uhh, I wanted to see the URI Exception on logs and if possible, even show it to the users somehow. I thought logging would be best option :)

Reply:
The caller can start with just the URI supplied to it. That's what I want to achieve :)

Reply:
Maybe an appropriate message will do?

Reply:
Sounds good.